python/meterpreter/meterpreter.py

```diff
@@ -494,7 +498,18 @@ def from_request(request):
 		elif url.startswith('http'):
 			proxy = packet_get_tlv(request, TLV_TYPE_TRANS_PROXY_HOST).get('value')
 			user_agent = packet_get_tlv(request, TLV_TYPE_TRANS_UA).get('value', HTTP_USER_AGENT)
-			transport = HttpTransport(url, proxy=proxy, user_agent=user_agent)
+			http_headers = packet_get_tlv(request, TLV_TYPE_TRANS_HEADERS).get('value', None)
+			if http_headers:
+				headers = {}
+				for h in http_headers.strip().split("\r\n"):
+					p = h.split(':')
+					headers[p[0].upper()] = ''.join(p[1:0])
+				http_host = headers.get('HOST')
+				http_cookie = headers.get('COOKIE')
+				http_referer = headers.get('REFERER')
+
+			transport = HttpTransport(url, proxy=proxy, user_agent=user_agent, http_host=http_host,
```

Comment:
Is it intentional that the other `headers` don't go anywhere? Seems like `HttpTransport` could take the `headers` dictionary and remove the special values out, process them as necessary and still add the remaining values.

Reply:
I think shifting this to a dictionary is probably a good idea. So I'll add user agent to that dictionary as well. You cool with that?

Reply:
Yup :+1: